usage-examples/go/atlas-sdk-go/internal/billing/collector.go

```diff
@@ -0,0 +1,134 @@
+package billing
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"atlas-sdk-go/internal/errors"
+
+	"go.mongodb.org/atlas-sdk/v20250219001/admin"
+)
+
+// Detail represents the transformed billing line item
+type Detail struct {
+	Org      OrgInfo     `json:"org"`
+	Project  ProjectInfo `json:"project"`
+	Cluster  string      `json:"cluster"`
+	SKU      string      `json:"sku"`
+	Cost     float64     `json:"cost"`
+	Date     time.Time   `json:"date"`
+	Provider string      `json:"provider"`
+	Instance string      `json:"instance"`
+	Category string      `json:"category"`
+}
+
+// OrgInfo contains organization identifier information
+type OrgInfo struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// ProjectInfo contains project identifier information
+type ProjectInfo struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// CollectLineItemBillingData retrieves all pending invoices for the specified organization,
+// transforms them into detailed billing records, and filters out items processed before lastProcessedDate.
+// Returns a slice of billing Details or an error if no valid invoices or line items are found.
+func CollectLineItemBillingData(ctx context.Context, sdk admin.InvoicesApi, orgSdk admin.OrganizationsApi, orgID string, lastProcessedDate *time.Time) ([]Detail, error) {
+	req := sdk.ListPendingInvoices(ctx, orgID)
+	r, _, err := req.Execute()
+
+	if err != nil {
+		return nil, errors.FormatError("list pending invoices", orgID, err)
+	}
+	if r == nil || !r.HasResults() || len(r.GetResults()) == 0 {
```

Comment:
This is the spot where we're saying that no results = error, and returning that to the `line_items/main.go` call site, so the current iteration of this is still returning an error when there are no results. This is the spot where we would need to change the logic to avoid making this an error condition, not in `line_items/main.go`.

Reply:
derp! updated 😮‍💨